onthisday/__init__.py

```diff
@@ -0,0 +1,216 @@
+
+
+def get_years(years):
+    return set(years + tuple(map(lambda s: s.zfill(4), years)))
+
+
+def date_suffix(number) -> str:
+    number = int(number)
+    suffixes = {0: "th", 1: "st", 2: "nd", 3: "rd"}
+    for i in range(4, 10):
+        suffixes[i] = "th"
+    return str(number) + suffixes[int(str(number)[-1])]
+
+
+def now() -> datetime.datetime:
+    return datetime.datetime.now()
+
+
+class DateConverter(Converter):
+    """Date converter which uses dateparser.parse()."""
+
+    async def convert(self, ctx: Context, arg: str) -> datetime.datetime:
+        parsed = dateparser.parse(arg)
+        if parsed is None:
+            raise BadArgument("Unrecognized date/time.")
+        if parsed.strftime("%Y") != str(now().strftime("%Y")):
+            raise BadArgument("Please do not specify a year.")
+        return parsed
+
+
+class OnThisDay(commands.Cog):
+    """Find out what happened today, in multiple different years in history."""
+
+    __version__ = "1.0.0"
+    __author__ = "Kreusada"
+
+    def __init__(self, bot: Red):
+        self.bot = bot
+        self.session = aiohttp.ClientSession()
+        if 719988449867989142 in self.bot.owner_ids:
+            with contextlib.suppress(RuntimeError, ValueError):
+                self.bot.add_dev_env_value(self.__class__.__name__.lower(), lambda x: self)
+
+    def format_help_for_context(self, ctx: commands.Context) -> str:
+        context = super().format_help_for_context(ctx)
+        return f"{context}\n\nAuthor: {self.__author__}\nVersion: {self.__version__}"
+
+    def cog_unload(self):
+        self.bot.loop.create_task(self.session.close())
+        if 719988449867989142 in self.bot.owner_ids:
+            with contextlib.suppress(KeyError):
+                self.bot.remove_dev_env_value(self.__class__.__name__.lower())
+
+    async def red_delete_data_for_user(self, **kwargs):
+        """Nothing to delete"""
+        return
+
+    async def run_otd(
+        self, ctx: commands.Context, month_number, date_number, year, _random: bool = False
+    ):
+        """Oh my, is this scruffy code... but it WORKS!"""
+        try:
+            async with self.session.get(
+                ENDPOINT.format(f"{month_number}/{date_number}")
+            ) as session:
+                content = await session.json()
+        except aiohttp.ClientError as e:
+            log.exception(e)
+            return await ctx.maybe_send_embed(
+                warning("An error occured whilst retrieving information for this day.")
+            )
+        else:
+            wikipedia_for_the_day = content["wikipedia"]
+            events = filter(lambda x: retrieve_above_0(x["year"]), content["events"])
+            data = {
+                e["year"]: {"content": e["description"], "wikipedia": e["wikipedia"]}
+                for e in events
+            }
+            if _random:
+                _c = random.choice(list(data.keys()))
+                event = data[_c]
+                result = _c
```

Comment:
I don't see a reason for this line here

Reply:
Haha, I had to do that to avoid an UnlocalBoundError earlier, could've just named that variable `result` originally, thanks.